### PR TITLE
Preserve original folder structure. Fix output dir malformations. Fix process crash.

### PR DESCRIPTION
**Pull request by Tylerian**
This pull request addresses three different issues:
- Fixes a bug that crashed the process when a file failed to write.
- Fixes a bug that malformed the output dir when -o option is provided.
- Adds a feature to preserve the original folder structure of a source map.

### CODE REVIEW

File: lib/write-sources.js [hunk 1]
```diff
@@ -35,7 +36,12 @@ const writeSources = (filename, content, outdir, options) => {
     console.error(`File "${outputFilepath}" already exists, skipping!`);
   }
   else {
-    fs.writeFileSync(outputFilepath, content, 'utf8');
+    try {
+      fs.writeFileSync(outputFilepath, content, 'utf8');
+    }
+    catch (error) {
+      console.log(`Error while trying to write file "${outputFilepath}", skipping...`, error);
```
[the remark this answers is not in the document]
Comment (Tylerian) [reply lib/write-sources.js]: console is a NodeJS global...

File: README.md [hunk 2]
```diff
@@ -32,13 +32,13 @@ shuji - Reverse engineering JavaScript and CSS sources from sourcemaps
...
 
   -h, --help               Help and usage instructions
-  -V, --version            Version number
-  -v, --verbose            Verbose output, will print which file is currently being
-                           processed
   -o, --output-dir String  Output directory - default: .
+  -p, --preserve           Preserve sourcemap's original folder structure.
   -M, --match String       Regular expression for matching and filtering files -
                            default: \.map$
-  -r, --recursive          Recursively search matching files
```
Comment (paazmaya) [inline README.md]: Indeed, v0.6.0 removed that option

File: lib/write-sources.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (paazmaya) [reply lib/write-sources.js]: This try/catch is a good addition

**Comment by paazmaya** [comment]
I took two commits from here, README update and
> Fixes a bug that crashed the process when a file failed to write.

**Comment by paazmaya** [comment]
Similar functionality now released in `v0.8.0` via #39

**Pull request closed without merging**